pkg/gofr/datasource/pubsub/nats/connection_manager.go

```diff
@@ -0,0 +1,176 @@
+package nats
+
+import (
+	"context"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"gofr.dev/pkg/gofr/datasource"
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+//go:generate mockgen -destination=mock_jetstream.go -package=nats github.com/nats-io/nats.go/jetstream JetStream,Stream,Consumer,Msg,MessageBatch
+
+const (
+	ctxCloseTimeout = 5 * time.Second
+)
+
+type ConnectionManager struct {
+	conn             ConnInterface
+	jetStream        jetstream.JetStream
+	config           *Config
+	logger           pubsub.Logger
+	natsConnector    NATSConnector
+	jetStreamCreator JetStreamCreator
+}
+
+func (cm *ConnectionManager) JetStream() (jetstream.JetStream, error) {
+	if cm.jetStream == nil {
+		return nil, errJetStreamNotConfigured
```

Comment:
The method is exported, so the error should be too, except of the method could be unexported

Reply:
sure made it unexported